docs/modelview/modelview.rst

```diff
@@ -324,6 +324,52 @@ Another detail specific to ``PathView`` worth noticing is the usage of the attac
 
 When transforming images or other complex elements on in ``PathView``, a performance optimization trick that is common to use is to bind the ``smooth`` property of the ``Image`` element to the attached property ``PathView.view.moving``. This means that the images are less pretty while moving but smoothly transformed when stationary. There is no point spending processing power on smooth scaling when the view is in motion, as the user will not be able to see this anyway.
 
+Table Models
+------------
+
+All views discussed until now present an array of items one way or another. Even the ``GridView`` expects the model to provide a one dimensional list of items. For two dimensional tables of data you need to use the ``TableView`` element.
+
+Conceptually, the ``TableView`` works just as all other views. It combines a ``model`` with a ``delegate`` to form a grid. If given a list type model it displays a single column, making it very similar to the ``ListView`` element. However, for table style models, it becomes more powerful. 
+
+In the example below, we setup a simple ``TableView`` with a custom model exposed from C++. At the moment, it is not possible to create table style models directly from QML, but in the 'Qt and C++' chapter the concept is explained. The running example is shown in the figure below.
+
+.. figure:: assets/tableview.png
+
+Before we can use the ``TableView`` element, we must make sure that the ``2.12`` version of ``QtQuick`` is imported. After that, we can set it up. In this example below, we set the ``rowSpacing`` and ``columnSpacing`` to control the horizontal and vertical gaps between delegates. The rest of the properties are setup as for any other type of view.
+
+.. literalinclude:: src/tableview/main.qml
+    :start-after: M1>>
+    :end-before: <<M1
+
+.. literalinclude:: src/tableview/main.qml
+    :start-after: M2>>
+    :end-before: <<M2
+    
+The delegate itself can carry an implicit size through the ``implicitWidth`` and ``implicitHeight``. This is what we do in the example below. The actual data contents, i.e. ``display``, depends on what the model exposes.
```

Comment:
i.e. the string returned from the model's ``display`` role

Reply:
Fixed